packages/neuron-wallet/src/keys/hd.ts

```diff
@@ -0,0 +1,104 @@
+import bip32, { BIP32 } from 'bip32'
+import { Child, KeysData } from './keystore'
+import Address from './address'
+
+enum BIP44Params {
+  Purpose = "44'",
+  // 360 is tentative value
+  CoinTypeTestnet = "360'",
+  Account = "0'",
+}
+
+enum AddressType {
+  Receive = 0,
+  Change = 1,
+}
+
+const HD = {
+  generateReceiveAndChangeAddresses: (keysData: KeysData, receiveNumber: number, changeNumber: number) => {
+    if (receiveNumber < 1 || changeNumber < 1) {
```

Comment:
What does `receiveNumber` mean? And shouldn't we guard program so that we don't allow a big number to pass in and make a long time loop?

Reply:
A new wallet will have 20 addresses which includes 17 receive addresses and 3 change addresses. 

So do you think this action should be done by controller or ui ?